src/Nest/Aggregations/AggregateDictionary.cs

```diff
@@ -1,40 +1,42 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
-using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Nest
 {
-	internal class AggregateDictionaryConverter : VerbatimDictionaryKeysJsonConverter<string, IAggregate>
-	{
-		public override bool CanRead => true;
-
-		public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
-		{
-			var dict = serializer.Deserialize<Dictionary<string, IAggregate>>(reader);
-			return new AggregateDictionary(dict);
-		}
-	}
-
 	/// <summary>
 	/// Contains aggregates that are returned by Elasticsearch. In NEST `Aggregation` always refers to an aggregation
 	/// going to elasticsearch and an `Aggregate` describes an aggregation going out.
 	/// </summary>
 	[ContractJsonConverter(typeof(AggregateDictionaryConverter))]
 	public class AggregateDictionary : IsAReadOnlyDictionaryBase<string, IAggregate>
 	{
+		private static string TypedKeysTokens(string key)
+		{
+//typed_keys = true on results in aggregation keys being returned as "<type>#<name>"
+			var tokens = key.Split(TypedKeysSeparator, 2, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
var tokens = TypedKeyTokens(key); ?

Reply:
whole method can be remove alltogether 👍